components/admin_console/team_channel_settings/team/details/team_details.jsx

```diff
@@ -111,6 +118,27 @@ export default class TeamDetails extends React.PureComponent {
         let saveNeeded = false;
 
         const {team, groups: origGroups, teamID, actions} = this.props;
+        if (this.teamToBeArchived()) {
+            const result = await actions.deleteTeam(team.id);
+            if ('error' in result) {
+                serverError = <FormError error={result.error.message}/>;
+                saveNeeded = true;
```

Comment:
Can we have this instance of `saveNeeded` scoped to this block, since we return at the end of the block? This way there's no confusion on whether "changing" that value here could affect code below this block.
```suggestion
        if (this.teamToBeArchived()) {
            let saveNeeded = false; // add this line here
            const result = await actions.deleteTeam(team.id);
            if ('error' in result) {
                serverError = <FormError error={result.error.message}/>;
                saveNeeded = true;
```